### DISCUSSION TITLE
Windows arm-zephyr-eabi GDB requires Python 3.8

### DISCUSSION THREAD

**Discussion by skalldri** [zephyrproject-rtos/sdk-ng#514] [Q&A] [answered]
Hi there,

The currently released Zephyr SDK for Windows (0.14.2) requires Python 3.8 be installed for certain components to work. So far, I've found that the arm-zephyr-eabi GDB instance needs access to `python38.dll` to function.

The most recent version of Zephyr asks users to install Python 3.10 (or perhaps more accurately, Python 3.10 is what ships with Chocolatey by default on Windows currently). Python 3.10 isn't compatible with the released Zephyr SDK, it needs Python 3.8. This means that users need to install two versions of Python side-by-side to use Zephyr and it's SDK.

Could this potentially be fixed, so that Zephyr and the SDK use the same version of Python? Or could the SDK ship with a re-distributable Python installer to avoid this issue?

**Comment by stephanosio** [comment, marked as answer]
Please see https://github.com/zephyrproject-rtos/sdk-ng/issues/407#issuecomment-943880289.

If you actually require the Python scripting capability, you can install the Python 3.8 by running the following command (there is a [PR to add this to the Getting Started Guide](https://github.com/zephyrproject-rtos/zephyr/pull/46949/commits/f12df71b19d5a856b0085640abf3003d9e1f0775)):
```
choco install python --version=3.8.0
```

Also from the upcoming Zephyr SDK release (0.15.0), the default GDB executable (`*-gdb`) will be built without the Python scripting support and no longer require `libpython`; instead, a separate Python scripting capable GDB variant (`*-gdb-py`) will be provided in case you actually need this feature.

**Reply by skalldri** [reply]
Thanks for the reply @stephanosio! Yes, I was able to fix this by installing Python 3.8 alongside Python 3.10. Glad to know that this is going to be fixed in the near future! Thank you!

**Comment by erwango** [comment]
@stephanosio Any new about an upcoming 3.10 version support ? (I guess this depends on gdb itself)
